sdk/typescript/src/transactions/plugins/NameResolutionPlugin.ts

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { SuiGraphQLClient } from '../../graphql/client.js';
+import type { NameResolutionRequest } from '../../utils/move_registry.js';
+import {
+	findTransactionBlockNames,
+	listToRequests,
+	replaceNames,
+} from '../../utils/move_registry.js';
+import type { BuildTransactionOptions } from '../json-rpc-resolver.js';
+import type { TransactionDataBuilder } from '../TransactionData.js';
+
+export type NameResolutionPlugin = {
```

Comment:
Probably should add Options to the end of this type

Reply:
Why do we need options (given this is an internal implementation, not visible to the user), and what would be its  type if we really needed it? Since this is an object-style input, can't we just introduce new keys in future versions if needed?

Reply:
I just mean rename this type to `NamedPackagesPluginOption`

Reply:
Ok now it makes sense 😆